packages/alfa-rules/src/sia-er62/diagnostics.ts

```diff
@@ -0,0 +1,203 @@
+
+import { Contrast } from "../../src/common/diagnostic/contrast";
+
+import { Serialise } from "./serialise";
+
+type Name = Property.Name | Property.Shorthand.Name;
+
+export class ElementDistinguishable
+  implements Equatable, Hashable, Serializable
+{
+  public static of(
+    style: Iterable<readonly [Name, string]> = [],
+    pairings: ReadonlyArray<Contrast.Pairing> = []
+  ): ElementDistinguishable {
+    return new ElementDistinguishable(Map.from(style), pairings);
+  }
+
+  private readonly _style: Map<Name, string>;
+  private readonly _pairings: ReadonlyArray<Contrast.Pairing>;
+
+  private constructor(
+    style: Map<Name, string>,
+    pairings: ReadonlyArray<Contrast.Pairing>
+  ) {
+    this._style = style;
+    this._pairings = pairings;
+  }
+
+  public get style(): Map<Name, string> {
+    return this._style;
+  }
+
+  public get pairings(): ReadonlyArray<Contrast.Pairing> {
+    return this._pairings;
+  }
+
+  public withStyle(
+    ...styles: ReadonlyArray<readonly [Name, string]>
+  ): ElementDistinguishable {
+    return ElementDistinguishable.of(
+      [...this._style, ...styles],
+      this._pairings
+    );
+  }
+
+  public equals(value: ElementDistinguishable): boolean;
+
+  public equals(value: unknown): value is this;
+
+  public equals(value: unknown): boolean {
+    return (
+      value instanceof ElementDistinguishable &&
+      value._style.equals(this._style) &&
+      Array.equals(value._pairings, this._pairings)
+    );
+  }
+
+  public hash(hash: Hash): void {
+    this._style.hash(hash);
+    Array.hash(this._pairings, hash);
+  }
+
+  public toJSON(): ElementDistinguishable.JSON {
+    return {
+      style: this._style.toJSON(),
+      pairings: Array.toJSON(this._pairings),
+    };
+  }
+}
+
+export namespace ElementDistinguishable {
+  export interface JSON {
+    [key: string]: json.JSON;
+    style: Map.JSON<Name, string>;
+    pairings: Array<Contrast.Pairing.JSON>;
+  }
+
+  export function serialise(
+    element: Element,
+    device: Device,
+    target: Element,
+    context: Context = Context.empty()
+  ): Iterable<readonly [Name, string]> {
+    const style = Style.from(element, device, context);
+
+    const border = (["color", "style", "width"] as const).map((property) =>
+      Serialise.borderShorthand(style, property)
+    );
+
+    const cursor = context.isHovered(target)
+      ? [["cursor", Serialise.getLonghand(style, "cursor")] as const]
+      : [];
+
+    return [
+      ...border,
+      ...cursor,
+      ["color", Serialise.getLonghand(style, "color")] as const,
+      ["font", Serialise.font(style)] as const,
+      [
+        "vertical-align",
+        Serialise.getLonghand(style, "vertical-align"),
+      ] as const,
+      ["background", Serialise.background(style)] as const,
+      ["outline", Serialise.outline(style)] as const,
+      ["text-decoration", Serialise.textDecoration(style)] as const,
+      // ["cursor", Serialise.getLonghand(style, "cursor")] as const,
```

Comment:
This I found commented out when moved, not sure if it's on purpose or a leftover